In [1]:
import pandas as pd  
import numpy as np  
import matplotlib.pyplot as plt  
import seaborn as sns

In [2]:
adult_lit = pd.read_csv('../data/UNdata_Export_20230926_021055618.csv', skipfooter = 2, engine = 'python')

In [3]:
adult_lit

Country or Area       Subgroup  Year                       Source  \
0       Afghanistan  Female 15+ yr  2000  UNESCO_UIS Database_Sep2007   
1       Afghanistan    Male 15+ yr  2000  UNESCO_UIS Database_Sep2007   
2           Albania  Female 15+ yr  2001  UNESCO_UIS Database_Sep2007   
3           Albania    Male 15+ yr  2001  UNESCO_UIS Database_Sep2007   
4           Algeria  Female 15+ yr  2002  UNESCO_UIS Database_Sep2007   
..              ...            ...   ...                          ...   
475          Zambia    Male 15+ yr  1990  UNESCO_UIS Database_Sep2007   
476        Zimbabwe  Female 15+ yr  2004  UNESCO_UIS Database_Sep2007   
477        Zimbabwe  Female 15+ yr  1992  UNESCO_UIS Database_Sep2007   
478        Zimbabwe    Male 15+ yr  2004  UNESCO_UIS Database_Sep2007   
479        Zimbabwe    Male 15+ yr  1992  UNESCO_UIS Database_Sep2007   

        Unit  Value  Value Footnotes  
0    Percent   12.6              NaN  
1    Percent   43.1              NaN  
2    Percent   98.3              NaN  
3    Percent   99.2              NaN  
4    Percent   60.1              NaN  
..       ...    ...              ...  
475  Percent   73.0              NaN  
476  Percent   86.2              1.0  
477  Percent   78.5              NaN  
478  Percent   92.7              1.0  
479  Percent   88.9              NaN  

[480 rows x 7 columns]

In [4]:
adult_lit = adult_lit.drop(columns = ['Source', 'Value Footnotes', 'Unit'])

In [5]:
adult_lit = adult_lit.rename(columns = {'Value' : 'Percent_Value', 'Country or Area' : 'Country', 'Subgroup' : 'Gender'})

In [6]:
adult_lit

Country         Gender  Year  Percent_Value
0    Afghanistan  Female 15+ yr  2000           12.6
1    Afghanistan    Male 15+ yr  2000           43.1
2        Albania  Female 15+ yr  2001           98.3
3        Albania    Male 15+ yr  2001           99.2
4        Algeria  Female 15+ yr  2002           60.1
..           ...            ...   ...            ...
475       Zambia    Male 15+ yr  1990           73.0
476     Zimbabwe  Female 15+ yr  2004           86.2
477     Zimbabwe  Female 15+ yr  1992           78.5
478     Zimbabwe    Male 15+ yr  2004           92.7
479     Zimbabwe    Male 15+ yr  1992           88.9

[480 rows x 4 columns]

**Which Country had the highest Adult literacy Rate and in which year?**

In [43]:
adult_lit.sort_values('Percent_Value', ascending = False)

Country         Gender  Year  Percent_Value
147       Estonia    Male 15+ yr  1989           99.9
216        Latvia    Male 15+ yr  2000           99.8
217        Latvia    Male 15+ yr  1989           99.8
120          Cuba  Female 15+ yr  2002           99.8
121          Cuba    Male 15+ yr  2002           99.8
..            ...            ...   ...            ...
86           Chad  Female 15+ yr  2000           12.8
0     Afghanistan  Female 15+ yr  2000           12.6
257          Mali  Female 15+ yr  1998           11.9
63   Burkina Faso  Female 15+ yr  1991            8.2
62   Burkina Faso  Female 15+ yr  1996            8.1

[480 rows x 4 columns]

**The country with the highest adult literacy was Estonia is 1989 with 99.9% of males 15+**

In [56]:
# Average adult literacy rate for each gender in a Country
adult_lit_pivot = adult_lit.pivot_table(values = "Percent_Value", index = "Country", columns = 'Gender').reset_index()

**How does the adult literacy rate differ between females and males**

In [57]:
adult_lit_pivot

Gender      Country  Female 15+ yr  Male 15+ yr
0       Afghanistan          12.60        43.10
1           Albania          98.30        99.20
2           Algeria          47.95        71.50
3            Angola          54.20        82.90
4         Argentina          96.60        96.70
..              ...            ...          ...
130       Venezuela          90.85        92.00
131         Vietnam          84.85        93.45
132           Yemen          25.90        64.90
133          Zambia          58.60        74.65
134        Zimbabwe          82.35        90.80

[135 rows x 3 columns]

**Which country had year the highest % change in Adult literacy rate? Create a line plot showing this country's adult literacy rate. Create another showing the country with the second highest % change in adult literacy. How do the trends in these countries compare?**  

In [10]:
adult_lit['Percent_Value'].sort_values(ascending = False)

147    99.9
216    99.8
217    99.8
120    99.8
121    99.8
       ... 
86     12.8
0      12.6
257    11.9
63      8.2
62      8.1
Name: Percent_Value, Length: 480, dtype: float64

In [11]:
GDP_df = pd.read_csv('../data/gdp_per_capita.csv')

In [12]:
GDP_df = GDP_df.drop(['Value Footnotes'], axis = 1)

In [13]:
GDP_df = GDP_df.rename(columns = {'Country or Area' : 'Country', 'Value' : 'GDP_Per_Capita'})

In [14]:
adult_lit_and_GDP = pd.merge(GDP_df, adult_lit)

In [15]:
adult_lit_and_GDP

Country  Year  GDP_Per_Capita         Gender  Percent_Value
0     Albania  2001     6441.449833  Female 15+ yr           98.3
1     Albania  2001     6441.449833    Male 15+ yr           99.2
2     Algeria  2002     9232.415784  Female 15+ yr           60.1
3     Algeria  2002     9232.415784    Male 15+ yr           79.6
4      Angola  2001     4746.660523  Female 15+ yr           54.2
..        ...   ...             ...            ...            ...
355    Zambia  1990     2190.181724    Male 15+ yr           73.0
356  Zimbabwe  2004     1979.200783  Female 15+ yr           86.2
357  Zimbabwe  2004     1979.200783    Male 15+ yr           92.7
358  Zimbabwe  1992     2437.152908  Female 15+ yr           78.5
359  Zimbabwe  1992     2437.152908    Male 15+ yr           88.9

[360 rows x 5 columns]

In [ ]:
adult_lit_and_GDP['Gender']==

In [20]:
adult_lit_Fem = adult_lit_and_GDP.loc[adult_lit_and_GDP['Gender'].isin(['Female 15+ yr'])].pivot_table(values = 'Percent_Value', index = 'Year', columns = 'Gender').dropna().reset_index()

In [21]:
adult_lit_Fem

Gender  Year  Female 15+ yr
0       1990      73.860000
1       1991      61.605882
2       1992      73.844444
3       1993      76.066667
4       1994      66.400000
5       1995      81.180000
6       1996      69.471429
7       1997      25.000000
8       1998      37.975000
9       1999      88.250000
10      2000      74.509677
11      2001      81.516000
12      2002      68.718182
13      2003      54.685714
14      2004      76.203846
15      2005      64.690000

In [23]:
adult_lit_Male = adult_lit_and_GDP.loc[adult_lit_and_GDP['Gender'].isin(['Male 15+ yr'])].pivot_table(values = 'Percent_Value', index = 'Year', columns = 'Gender').dropna().reset_index()

In [24]:
adult_lit_Male

Gender  Year  Male 15+ yr
0       1990    84.050000
1       1991    73.517647
2       1992    84.144444
3       1993    84.600000
4       1994    76.112500
5       1995    87.300000
6       1996    76.542857
7       1997    54.800000
8       1998    59.100000
9       1999    90.566667
10      2000    84.025806
11      2001    87.764000
12      2002    79.336364
13      2003    65.528571
14      2004    85.365385
15      2005    77.870000

**Does Adult Literacy and GDP have a correlation**

**Which continent has the highest Adult Literacy**